ledger/src/blockstore_processor.rs

```diff
@@ -3548,13 +3556,42 @@ pub mod tests {
 
             let entry = next_entry(&bank.last_blockhash(), 1, vec![tx]);
             let bank = Arc::new(bank);
-            let _result = process_entries_for_tests_without_scheduler(&bank, vec![entry]);
+            let result = process_entries_for_tests_without_scheduler(&bank, vec![entry]);
+            assert!(result.is_ok()); // No failing transaction error - only instruction errors
             bank.freeze();
 
+<<<<<<< HEAD
             assert_eq!(blockhash_ok, bank.last_blockhash());
             assert!(bankhash_ok != bank.hash());
             if let Some(bankhash) = bankhash_err {
                 assert_eq!(bankhash, bank.hash());
+=======
+            // Transaction success/failure should not affect block hash ...
+            assert_eq!(
+                ok_bank_details
+                    .bank_hash_components
+                    .as_ref()
+                    .unwrap()
+                    .last_blockhash,
+                bank_details
+                    .bank_hash_components
+                    .as_ref()
+                    .unwrap()
+                    .last_blockhash
+            );
+            // AND should not affect bankhash IF the rent is collected during freeze.
+            assert_eq!(ok_bank_details == bank_details, fee_payer_in_rent_partition);
```

Comment:
This is the assertion we need to do as noted by what we have in master:
https://github.com/anza-xyz/agave/blob/1e721f0862a5656b9a8651cecc63008f9f02b000/ledger/src/blockstore_processor.rs#L3660
which you can also see is what was added in the initial PR:
https://github.com/anza-xyz/agave/pull/3916/files#diff-281d947e593a76f3c3380804162dc8cf426fcc33e0ae1192589c8193221cd1e2R3626